lib/modules/manager/bundler/artifacts.ts

```diff
@@ -43,6 +43,21 @@ function buildBundleHostVariable(hostRule: HostRule): Record<string, string> {
   };
 }
 
+const resolvedPkgRegex = regEx(
+  /(?<pkg>\S+)(?:\s*\([^)]+\)\s*)? was resolved to/g
+);
+
+function getResolvedPackages(input: string): string[] {
+  const result: string[] = [];
+  let resolveMatchGroups = resolvedPkgRegex.exec(input)?.groups;
+  while (resolveMatchGroups) {
```

Comment:
Any chance this could enter an infinite loop? Maybe check if `pkg` is already in `result` and break out of the loop if so?

Reply:
Checking for infinite loop is always a good idea 👍

Reply:
I'm not sure about infinite recursion criteria, so instead I've limited nested calls by 10 at maximum

Reply:
So it's impossible that this `while` loop won't go infinitely? What about checking for a repeat of same `pkg`?

Reply:
From iteration to iteration, we extend `updatedDeps` list with at least one new package name which is obtained from the ouside source. Seems like depth limit is the best we can do here.

Reply:
i don't see this resolved, see my other comment

Reply:
Oh sorry, confused this with another potential infinite loop case in this PR